api/array.hpp

```diff
@@ -142,7 +142,7 @@ namespace jule
 
         constexpr jule::Bool operator==(const jule::Array<Item, N> &src) const
         {
-            if (*this == src)
+            if (this == &src)
```

Comment:
What about `this.buffer == src.buffer`?

Reply:
Still this can be only true if `this` and `&src` are the same. 

If `this` and `src` are different objects, then their buffers are in different places in memory.

Reply:
I think my suggestion is more clear about what the condition is for.
What do you think about this, should we follow my suggestion or just merge PR?

Reply:
`this == &src` is more direct.